tests/unit/test_database_memory.py

```
@@ -65,6 +65,475 @@
+class TestIsMemoryDb:
+    """Comprehensive tests for the _is_memory_db function."""
+
+    def test_is_memory_db_with_literal_memory(self, monkeypatch, restore_database_module):
+        """Test that _is_memory_db returns True for literal ':memory:' string."""
+        monkeypatch.setenv("DATABASE_URL", "sqlite:///:memory:")
+        reloaded_database = importlib.reload(database)
+        
+        assert reloaded_database._is_memory_db() is True
+        assert reloaded_database._is_memory_db(":memory:") is True
+
+    def test_is_memory_db_with_file_uri_memory(self, monkeypatch, restore_database_module):
```

Ruff (0.14.4)

Unused method argument: `monkeypatch`

(ARG002)